starter_code/app.js

```diff
@@ -1,28 +1,87 @@
-require('dotenv').config()
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
-
-
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
-
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
 
-
-
-
-
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => {
+    spotifyApi.setAccessToken(data.body["access_token"]);
+  })
+  .catch(error => {
+    console.log("Something went wrong when retrieving an access token", error);
+  });
+
+app.get("/", (req, res) => {
+  res.render("home.hbs");
+});
 
 // the routes go here:
 
-
-
-app.listen(3000, () => console.log("My Spotify project running on port 3000 🎧 🥁 🎸 🔊"));
+app.get("/artists", (req, res) => {
+  let artistRequest = req.query.artistName;
+
+  spotifyApi
+    .searchArtists(artistRequest)
+    .then(data => {
+      console.log("The received data from the API: ", data.body.artists.items);
+      // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+      //   res.send(data.body);
+      let artistsArr = data.body.artists.items;
+      res.render("artists", { artistList: artistsArr });
+    })
+    .catch(err => {
+      console.log("The error while searching artists occurred: ", err);
+    });
+});
+
+app.listen(3000, () =>
```

Comment:
I would place this at the end of your code, not in the middle of all the routes. 

To make your code cleaner, have the routes first and then the app.listen()